packages/core/src/generators/nuget-reference/generator.ts

```diff
@@ -34,7 +34,7 @@ export default async function (
     resolvedVersion !== options.version &&
     resolvedVersion !== ALLOW_MISMATCH
   ) {
-    options.version = resolvedVersion;
+    params.version = resolvedVersion;
```

Comment:
On line 20, do you care to add allowVersionMismatch to the destructure? It shouldn't be passed to `dotnet`.

Thats a leftover issue from my impl, and otherwise this looks good. thanks!